src/dmd/root/array.d

```diff
@@ -178,41 +178,42 @@ public:
     Array!T* copy() const nothrow
     {
         auto a = new Array!T();
-        a.setDim(dim);
-        memcpy(a.data, data, dim * (void*).sizeof);
+        a.setDim(length);
+        memcpy(a.data, data, length * (void*).sizeof);
         return a;
     }
 
     void shift(T ptr) nothrow
     {
         reserve(1);
-        memmove(data + 1, data, dim * (*data).sizeof);
+        memmove(data + 1, data, length * (*data).sizeof);
         data[0] = ptr;
-        dim++;
+        length++;
     }
 
     void zero() nothrow pure
     {
-        data[0 .. dim] = T.init;
+        data[0 .. length] = T.init;
     }
 
     T pop() nothrow pure
     {
-        return data[--dim];
+        return data[--length];
     }
 
     extern (D) inout(T)[] opSlice() inout nothrow pure
     {
-        return data[0 .. dim];
+        return data[0 .. length];
     }
 
     extern (D) inout(T)[] opSlice(size_t a, size_t b) inout nothrow pure
     {
-        assert(a <= b && b <= dim);
+        assert(a <= b && b <= length);
         return data[a .. b];
     }
 
     alias opDollar = dim;
+    alias length = dim;
```

Comment:
Isn't it better to rename `dim` to `length` and add the alias the other way around?

Reply:
sure why not